pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/ReadableTestBase.java

```diff
@@ -1293,12 +1334,30 @@ void readZigZag() {
             assertThat(seq.position()).isEqualTo(pos + 2);
         }
 
+        @Test
+        @DisplayName("Reading a varint that is not properly encoded throws DataEncodingException")
+        void readInvalidVarInt() {
+            // Given a very long sequence of bytes all with the "continuation" bit set
+            final var seq = sequence(new byte[] {
+                    (byte) 0b10101101,
+                    (byte) 0b10000010,
+                    (byte) 0b10000010,
+                    (byte) 0b10000010,
+                    (byte) 0b10000010,
+                    (byte) 0b10000010,
+                    (byte) 0b10000010,
+                    (byte) 0b10000010,
+                    (byte) 0b10000010,
+                    (byte) 0b10000010
+            });
+            // When we try to decode an int, the lack of a "terminator" bit causes a DataEncodingException
+            assertThatThrownBy(() -> seq.readVarInt(false)).isInstanceOf(DataEncodingException.class);
+        }
```

Comment:
Is there a test that calls `readVarInt()` for a sequence of bytes, where more than 5 bytes have continuation bit set, and the last one does not? `readVarLong()` should handle such a sequence just fine, but (to my surprise), `readVarInt()` should not fail either. It should read it as a long and then downcast to int, if I remember correctly.

Reply:
Yes, readVarInt() basically delegates to readVarLong(). So there's no way to meaningfully test this until/unless we actually separate them and start failing the readVarInt on longer sequences. But I guess this would introduce some incompatibilities with Google's implementation (IIRC, right?), that's why we simply delegate for now.